### ISSUE TITLE
we need lightbdd events

### ISSUE THREAD

**Issue by Vikaskumargd** [LightBDD/LightBDD#142]
In Specflow we can use the these events before the test execution

 [BeforeFeature]

        public static void BeforeFeature()
        {
            FeatureContext.Current.Set<ExtentReports>(logger.extent);

        }

Is their any events in the lightbdd we can write  before the execution of the test case

**Comment by Suremaker** [comment]
Hello @Vikaskumargd ,

I presume you have found the answer because you have closed the ticket, but just in case you did not, let me give some hints.

First of all, LightBDD works on top of the underlying test framework, so if the framework you are using supports it, you can freely use that pattern for LightBDD tests. For example, if your tests are using LightBDD.NUnit3, the `before feature/after feature` code can be added by just adding methods annotated with `[OneTimeSetUp]/[OneTimeTearDown]` attributes, described on [nunit wiki page](https://github.com/nunit/docs/wiki/OneTimeSetUp-Attribute).

LightBDD itself offers a few ways of executing the additional code as well, however it is more Scenario oriented (I'll explain the reason further in the response).

If you are interested to execute some code around the tests, those would be the best ways to do it purely in LightBDD:
* you can use [Scenario Decorators](https://github.com/LightBDD/LightBDD/wiki/Extending-Test-Behavior#scenario-decorators) for wrapping scenario method, or [Step Decorators](https://github.com/LightBDD/LightBDD/wiki/Extending-Test-Behavior#step-decorators) to wrap each step method with additional code; both decorator types can be applied declaratively on the specific method or globally on all (details are on [wiki page](https://github.com/LightBDD/LightBDD/wiki/Extending-Test-Behavior#applying-decorators)),
* you can run code once (globally) before any tests or after all tests by overriding `OnSetUp()/OnTearDown()` methods of `LightBDDScopeAttribute` (details are on [wiki page](https://github.com/LightBDD/LightBDD/wiki/LightBDD-Configuration)).

There is no similar hook to above ones for executing code once, before any or after all tests within one class/feature as most of the frameworks does not execute tests against the same instance of the class (nunit is the exception and as mentioned above, you can just use `[OneTimeSetUp]/[OneTimeTearDown]` attributes there). What this means is that such hooks won't be able to influence feature class instance fields in any way except the static methods/fields of the class.

If you are just looking for the hooks triggering some code before of after feature (while not being interested in the feature class instance), then you can implement own progress notifier that will offer you such hooks. The progress notifiers configuration is described on wiki [here](https://github.com/LightBDD/LightBDD/wiki/Test-Progress-Notification#customizing-progress-notification), where the interface that you would need to implement will be: [IFeatureProgressNotifier](https://github.com/LightBDD/LightBDD/blob/master/src/LightBDD.Core/Notification/IFeatureProgressNotifier.cs).

Finally, if you are interested to provide some instance for scenarios in the feature class, you may be interested also in reading about DI support in LightBDD: [wiki page](https://github.com/LightBDD/LightBDD/wiki/DI-Containers).

-----
To summarize, I am not sure what exactly would you like to achieve with such events, but the majority of cases when people wanted to run some additional code around the tests were about making a scenario or step decorator and apply it on globally on all methods or locally on specific method.

Please let me know if that helped or if you have any further questions.

**Comment by Suremaker** [comment]
Closed due to inactivity.
